ir/src/integrity_stmts/graph.rs

```diff
@@ -117,10 +114,13 @@ impl AlgebraicGraph {
                 let node_index = self.insert_op(Operation::RandomValue(index));
                 Ok((trace_segment, node_index, CURRENT_ROW))
             }
-            IntegrityExpr::IndexedTraceAccess(column_access) => {
+            Expression::IndexedTraceAccess(column_access) => {
                 self.insert_trace_access(symbol_table, column_access)
```

Comment:
If we have `insert_named_trace_access`, may be we should rename this function to `insert_indexed_trace_access` to maintain consistency?

Reply:
Agreed! I was going to do it in a different PR, to minimize the amount of changes in this one. (I was trying to keep this to strictly the changes required for the new structure)